userspace/sysdig/chisels/spy_file.lua

```diff
@@ -93,7 +103,23 @@ function on_init()
 		filter = string.format("%s%s", filter, "evt.is_io=true and ")
 	end
 
-	filter = string.format("%s%s", filter, "fd.type=file and evt.dir=< and evt.failed=false")
+	if type_of_file == "F" or type_of_file == "f" then
+		filter = string.format("%s%s", filter, "fd.type=file and ")
+	elseif type_of_file == "P" or type_of_file == "p" then
+		filter = string.format("%s%s", filter, "fd.type=pipe and ")
+	elseif type_of_file == "U" or type_of_file == "u" then
+		filter = string.format("%s%s", filter, "fd.type=unix and ")
+	elseif type_of_file == "FP" or type_of_file == "fP" or type_of_file == "Fp" or type_of_file == "fp" or type_of_file == "PF" or type_of_file == "pF" or type_of_file == "Pf" or type_of_file == "pf" then
```

Comment:
For all these if/elseif I would use `type_of_file:lower() == "f"`, etc. instead of enumerating all possible cases.